StPicoDstMaker/StPicoDstMaker.cxx

```diff
@@ -561,7 +561,7 @@ void StPicoDstMaker::fillTracks()
   {
     StMuTrack* pTrk = (StMuTrack*)mMuDst->primaryTracks(i);
     if (!pTrk) continue;
-    if (pTrk->id() < 0 || pTrk->id() >= 50000)
+    if (pTrk->id() < 0)
```

Comment:
It shows that we are not going to have more than ```nTrk``` tracks regardless of details implementation details of MuDST. Compilers are very good at keeping records of return types and they will gladly optimise out these conditions where they are not effective. When removing the check you basically introduce an implicit contract that requires StMuTrack::id() to always return value smaller than nTrk (by having a smaller return type or whatever other means).